### ISSUE TITLE
Pop-up warning people that blocking is public information

### ISSUE THREAD

**Issue by Paarsec** [bluesky-social/social-app#5812]
### Describe the Feature

I think people should get a popup when blocking someone for the first time, telling them that blocklists are accessible by others.

I also think this fact should be a permanent addition in the little blurb on top of your blocklist for good measure. I believe this will avoid a lot of headaches in the future and increase people's safety.

I can't remember if the popup is already a thing as I've joined way too long ago, but in case it is, I think the text blurb above the user's blocklist would still be a wise addition. Thanks!

**Comment by surfdude29** [comment]
I agree that this could be improved. What do you think of the wording I proposed in #3699?

**"Blocking is public. Blocked accounts will be prevented from seeing your content and they cannot reply in your threads, mention you, or otherwise interact with you. You will not see their content."**

**Comment by Paarsec** [comment]
> I agree that this could be improved. What do you think of the wording I proposed in #3699?
I think the string "Blocking is public" as the first one works perfectly, well done. I hope it gets approved.

**Comment by ConnorKrammer** [comment]
@surfdude29 Putting my copyeditor hat on for a second—I actually think "Blocking is public" is not clear in the way we want it to be. I think there are basically two core problems:
- First, blocking is always public, insofar as it's an _act_ that occurs in a public space, and other people can infer that it has happened by observing interactions or seeing involved parties post about it. You can't expect that nobody will _ever_ know you have blocked someone. In this sense, the sentence doesn't update the user's mental model meaningfully.
- Second, and contra to the former point, blocking is *not* public at the level Bluesky users will be thinking about. A block on Bluesky is public at the level of technical implementation, and it's important that people know that in theory that data is accessible. But it's not actually public in terms of the site interface, which is how they actually experience "public" and "private" on the site. Saying that blocking is public but without updating their mental models as to what exactly this *means* risks confusion and anxiety that may actually cause people not to use the feature. Perhaps what this suggests to a user is that clicking the button will post an alert to everyone that they have done so.

I'm not sure you can actually fit into a few words what we want to express here. The idea does not have a shape that fits preexisting mental models closely enough that we can convey it both cleanly and in a highly compressed manner. The closest I personally can manage at the moment might be something like this:

> **Blocked accounts will be prevented from seeing your content and they cannot reply in your threads, mention you, or otherwise interact with you. You will not see their content.**
> (i) _Block lists are public information. Bluesky does not provide a way for others to view who you have blocked on the site, but 3rd-party clients may do so._

This wording could probably be iterated on further. I think "Block lists are public information" is not as precise as I would like, for example. Do consider the structure I used, though: instead of tacking on the bit about public state to a message which is otherwise somewhat orthogonal, give the warning its own space and let it *look* like a warning or information box.

It would also be ideal to provide a link that points to some canonical resource that explains what it means in greater detail, but I do not know if one exists.

Hope I conveyed what I wanted to clearly enough! Thanks for the hard work helping to improve the site for everyone.

**Comment by surfdude29** [comment]
@ConnorKrammer Thank you for your thoughtful and considered feedback and your suggested wording, and I apologise for only replying many months later.

I have just opened a new PR taking into account your suggestions (with a few changes):
- #7441
